### PR TITLE
Refactored signal to align with rc.3

### PR DESCRIPTION
**Pull request by guillemcordoba**

### CODE REVIEW

File: src/api/client.ts [hunk 1]
```diff
@@ -65,15 +65,16 @@ export class WsClient extends Emittery {
           // We have received a system signal, do nothing
           return;
         }
+        const encodedAppSignal = derializedSignal["App"];
```
Comment (Connoropolous) [inline src/api/client.ts]: You could use SignalType.App here

File: src/api/app/types.ts [hunk 2]
```diff
@@ -101,11 +101,17 @@ export const SignalType = {
...
 export type Signal = {
-  [SignalType.App]: [CellId, any];
+  [SignalType.App]: EncodedAppSignal;
   [SignalType.System]: unknown;
 };
+export type EncodedAppSignal = {
+  cell_id: CellId;
+  zome_name: string;
+  signal: Uint8Array;
+};
```
Comment (Connoropolous) [inline src/api/app/types.ts:167]: What about introducing PermaLinks comments through to the Rust side code? Bumping into quite a few serialization issues ongoingly

Comment (jost-s) [reply src/api/app/types.ts:167]: I'm not familiar with PermaLinks, what is that?

Comment (Connoropolous) [reply src/api/app/types.ts:167]: oops, just meant like a github permalink... like this. 
Screenshot 2023-01-18 at 9 08 02 PM

**Pull request merged**